hadoop-hdds/client/src/main/java/org/apache/hadoop/ozone/client/io/ECBlockReconstructedStripeInputStream.java

```diff
@@ -212,12 +212,14 @@ public synchronized void setRecoveryIndexes(Collection<Integer> indexes) {
   }
 
   private void init() throws InsufficientLocationsException {
+    initialized = false;
     if (decoder == null) {
       decoder = CodecUtil.createRawDecoderWithFallback(getRepConfig());
     }
     if (!hasSufficientLocations()) {
-      throw new InsufficientLocationsException("There are insufficient " +
-          "datanodes to read the EC block");
+      String msg = "There are insufficient datanodes to read the EC block";
+      LOG.debug(msg);
```

Comment:
sounds more like LOG.error than LOG.debug

Reply:
Generally the caller should log the exception higher up the stack. If you log at the source of the exception, you generally end up logging it twice. That is the reason I made this a debug log, but also because we are not certain this fix is the root cause of the problem we have seen, so it would be good to have the debug log incase we see the problem again.